plugin/vcs/gitlab/gitlab.go

```diff
@@ -182,7 +182,7 @@ func (provider *Provider) APIURL(instanceURL string) string {
 	return fmt.Sprintf("%s/%s", instanceURL, apiPath)
 }
 
-// TryLogin will try to login GitLab.
+// TryLogin will try to fetch the user info from the current OAuth content of GitLab.
 func (provider *Provider) TryLogin(ctx context.Context, oauthCtx common.OauthContext, instanceURL string) (*vcs.UserInfo, error) {
```

Comment:
The common logic can be extract into a helper method like fetchUserInfo. Then TryLogin and FetchUserInfo can be implemented on top of that method.

Reply:
BTW, this reviewing thread shows a classic example of the difference between public interface and private method.

Readability is always important. While for public interface, the focus is more on whether the **interface** is clean, while for the private method, the focus is on whether the **implementation** is clean.